tests/integration/views/test_request.py

```diff
@@ -693,6 +693,98 @@ def test_request_index_user_output_checker(airlock_client):
     assert approved_ids == {r4.id}
 
 
+def test_request_index_user_request_progress(airlock_client):
+    airlock_client.login(workspaces=["test_workspace"], output_checker=True)
+    other = factories.create_user("other")
+    default_checkers = factories.get_default_output_checkers()
+
+    def generate_files(reviewed=False, checkers_a=None, checkers_b=None):
+        if checkers_a is None:
+            checkers_a = default_checkers
+        if checkers_b is None:
+            checkers_b = default_checkers
+        return [
+            factories.request_file(
+                path="afile.txt", contents="a", approved=reviewed, checkers=checkers_a
+            ),
+            factories.request_file(
+                path="bfile.txt", contents="b", approved=reviewed, checkers=checkers_b
+            ),
+        ]
+
+    # submitted, no-one has reviewed
+    r0 = factories.create_request_at_status(
+        "test_workspace",
+        status=RequestStatus.SUBMITTED,
+        files=generate_files(),
+    )
+    # submitted, all files reviewed (but not completed)
+    r1 = factories.create_request_at_status(
+        "test_workspace1",
+        status=RequestStatus.SUBMITTED,
+        files=generate_files(
+            reviewed=True,
+            checkers_a=[airlock_client.user],
+            checkers_b=[airlock_client.user],
+        ),
+    )
+    # partially reviewed by someone else, no files reviewed
+    r2 = factories.create_request_at_status(
+        "other_workspace",
+        status=RequestStatus.PARTIALLY_REVIEWED,
+        files=generate_files(reviewed=True),
+    )
+    # partially reviewed by someone else, some files reviewed
```

Comment:
hmm, the Status name makes this comment weird to parse, this is wordier but clearer:
```suggestion
    # someone else has completed their review, making the request partially reviewed.
    # some files reviewed by the user.
```